pkg/loki/loki.go

```diff
@@ -280,8 +280,23 @@ func newDefaultConfig() *Config {
 	return defaultConfig
 }
 
+// RunOpts configures custom behavior for running Loki.
+type RunOpts struct {
+	// customConfigEndpointHandlerFn is the handlerFunc to be used by the /config endpoint.
+	// If empty, default handlerFunc will be used.
+	customConfigEndpointHandlerFn func(http.ResponseWriter, *http.Request)
```

Comment:
Should we export it to allow defining custom values? Without that, I do not see a way to change the behaviour outside of `loki` package.

Reply:
Oops, good point. I'll make it public then.